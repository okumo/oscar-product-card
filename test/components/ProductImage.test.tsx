import React from 'react'
import render from 'react-test-renderer'
import { ProductImage } from '../../src/components'
import { ProductCard } from '../../src/components/ProductCard'
import { product2 } from '../data/products'

describe('ProductImage', () => {
    test('debe mostrar la imagen del producto', () => {
        const wrapper = render.create(
            <ProductImage img={product2.img} className='custom-class' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe mostrar el componente con la imagen del producto', () => {
        const wrapper = render.create(
            <ProductCard product={product2}>
                {() => (<ProductImage />
                )}
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})