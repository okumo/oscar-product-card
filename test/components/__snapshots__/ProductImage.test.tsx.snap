// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe mostrar la imagen del producto 1`] = `
<img
  alt="Coffe Mug"
  className="productImg custom-class"
  src="./coffee-mug.png"
/>
`;
